eras/conway/impl/src/Cardano/Ledger/Conway/State/CertState.hs

```diff
@@ -0,0 +1,275 @@
+  }
+  deriving (Show, Eq, Generic)
+
+-- | Function that looks up the deposit for currently registered DRep
+lookupDepositVState :: VState era -> Credential 'DRepRole -> Maybe Coin
+lookupDepositVState vstate = fmap drepDeposit . flip Map.lookup (vstate ^. vsDRepsL)
+
+instance Default (VState era) where
+  def = VState def def (EpochNo 0)
+
+instance NoThunks (VState era)
+
+instance NFData (VState era)
+
+instance Era era => DecShareCBOR (VState era) where
+  type
+    Share (VState era) =
+      ( Interns (Credential 'Staking)
+      , Interns (Credential 'DRepRole)
+      , Interns (Credential 'HotCommitteeRole)
+      )
+  getShare VState {vsDReps, vsCommitteeState} =
+    (internsFromSet (foldMap drepDelegs vsDReps), fst (getShare vsDReps), getShare vsCommitteeState)
+  decShareCBOR (cs, cd, _) =
+    decode $
+      RecD VState
+        <! D (decodeMap (interns cd <$> decCBOR) (decShareCBOR cs))
+        <! D decNoShareCBOR
+        <! From
+
+instance Era era => DecCBOR (VState era) where
+  decCBOR = decNoShareCBOR
+
+instance Era era => EncCBOR (VState era) where
+  encCBOR VState {..} =
+    encode $
+      Rec (VState @era)
+        !> To vsDReps
+        !> To vsCommitteeState
+        !> To vsNumDormantEpochs
+
+instance ToJSON (VState era) where
+  toJSON = object . toVStatePair
+  toEncoding = pairs . mconcat . toVStatePair
+
+toVStatePair :: KeyValue e a => VState era -> [a]
+toVStatePair vs@(VState _ _ _) =
+  let VState {..} = vs
+   in [ "dreps" .= vsDReps
+      , "committeeState" .= vsCommitteeState
+      , "numDormantEpochs" .= vsNumDormantEpochs
+      ]
+
+-- ===================================
+-- VState
+
+vsDRepsL :: Lens' (VState era) (Map (Credential 'DRepRole) DRepState)
+vsDRepsL = lens vsDReps (\vs u -> vs {vsDReps = u})
+
+vsCommitteeStateL :: Lens' (VState era) (CommitteeState era)
+vsCommitteeStateL = lens vsCommitteeState (\vs u -> vs {vsCommitteeState = u})
+
+vsNumDormantEpochsL :: Lens' (VState era) EpochNo
+vsNumDormantEpochsL = lens vsNumDormantEpochs (\vs u -> vs {vsNumDormantEpochs = u})
+
+vsActualDRepExpiry :: Credential 'DRepRole -> VState era -> Maybe EpochNo
+vsActualDRepExpiry cred vs =
+  binOpEpochNo (+) (vsNumDormantEpochs vs) . drepExpiry <$> Map.lookup cred (vsDReps vs)
+
+csCommitteeCredsL ::
+  Lens' (CommitteeState era) (Map (Credential 'ColdCommitteeRole) CommitteeAuthorization)
+csCommitteeCredsL = lens csCommitteeCreds (\cs u -> cs {csCommitteeCreds = u})
+
+epochStateRegDrepL ::
+  ConwayEraCertState era => Lens' (EpochState era) (Map (Credential 'DRepRole) DRepState)
+epochStateRegDrepL = esLStateL . lsCertStateL . certVStateL . vsDRepsL
+
+class EraCertState era => ConwayEraCertState era where
+  certVStateL :: Lens' (CertState era) (VState era)
+
+mkConwayCertState :: VState era -> PState era -> DState era -> ConwayCertState era
+mkConwayCertState v p d =
+  ConwayCertState
+    { conwayCertVState = v
+    , conwayCertPState = p
+    , conwayCertDState = d
+    }
```

Comment:
You can actually make this function make more abstract (same with `mkShelleyCertState`)
```suggestion
mkConwayCertState :: ConwayEraCertState era => VState era -> PState era -> DState era -> CertState era
mkConwayCertState v p d =
  def
    & certVStateL .~ v
    & certPStateL .~ p
    & certDStateL .~ d
```
or even better
```suggestion
mkConwayCertState :: ConwayEraCertState era => VState era -> PState era -> DState era -> CertState era
mkConwayCertState v p d =
  mkShelleyCertState p d & certVStateL .~ v
```
This will come in handy for future eras